arch-snapshot/arch.md

```diff
@@ -48,9 +48,300 @@ only as detailed as to explain how the actors control the artifacts. The goal is
+    end
+    
+    Requestor-->ExeUnit: Finish computations
+    deactivate ExeUnit
+    Requestor->>Provider: Destroy Activity
+    destroy ExeUnit
+    Provider->>ExeUnit: Terminate ExeUnit
+  end
+  Requestor->>Provider: Terminate Agreement
+```
+
+#### 6. Terminate the Agreement or await the Agreement termination event from the Requestor
+
+The Agreement can be terminated when either party chooses to end it. The reasons for termination are outlined in the
+Agreement, and different market negotiation protocols may permit termination for various reasons. Below is
+a non-exhaustive list of potential causes for termination:
+- The Agreement expires if it was established for a fixed duration.
+- The Requestor no longer needs the resources or has completed the computations.
+- One of the parties violates the terms of the Agreement, such as:
+  - The Requestor fails to accept Debit Notes within the agreed timeframe.
+  - The Provider issues Debit Notes more frequently than agreed.
+  - The Requestor fails to make timely payments, particularly in cases involving mid-agreement payments.
+
+An Agent can terminate the Agreement using the market's REST API. It should also monitor Agreement events
+to detect if the other party terminates the Agreement.
+
+An Agent has the option to attach additional information outlining the reasons for termination when ending the
+Agreement. While this is not mandatory, it is encouraged as it can provide valuable context for the other party,
+serving as diagnostic information or for other purposes.
+
+#### 7. Send an Invoice summarizing the total cost of the Agreement
+
+Once the Agreement is terminated, the Provider Agent should send an Invoice to the Requestor summarizing the total costs
+incurred throughout the Agreement. This Invoice should reflect the cumulative costs from all Activities. In response,
+the Requestor must either accept or reject the Invoice. However, regardless of the acceptance status, payment is mandatory
+for the total amount indicated by the accepted Debit Notes, as their acceptance constitutes a binding commitment to pay.
+
+```mermaid
+flowchart LR
+Activity1((Activity 1)) --o D11[Debit Note 1] --> D12[Debit Note 2] -->|...| D13[Debit Note N-th] --> Invoice[Invoice]
+Activity2((Activity 2)) --o D21[Debit Note 1] --> D22[Debit Note 2] -->|...| D23[Debit Note N-th] --> Invoice[Invoice]
+Activity3((Activity 3)) --o D31[Debit Note 1] --> D32[Debit Note 2] -->|...| D33[Debit Note N-th] --> Invoice[Invoice]
+```
+
+#### 8. Monitor Payment API events for Invoice settlement and payment confirmation
+
+As previously mentioned, payments are not immediate for several reasons: they are not scheduled right away, and batching 
+may occur. Furthermore, blockchain transactions are not immediate and may take time to process. Therefore, the Provider
+Agent should monitor Payment events. This can be done by listening for status changes to Settled on Invoice and Debit Note
```

Comment:
Rather than describing the API, I'm much more interested in why the Agent should monitor the payment status. What should it do if it doesn't receive the payment? BTW, how would it know the relationship between individual payments it receives and the invoices / debit notes?